File: app/src/main/java/com/example/myittaroostockinventorymanger/local/Batch.java
package com.example.myittaroostockinventorymanger.local;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Date;

@Entity
public class Batch {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "batch_id")
    private long batchId;
    @ColumnInfo(name = "stock_id")
    private long stockId;
    @ColumnInfo(name = "original_price")
    private double originalPrice;
    @ColumnInfo(name = "sale_price")
    private double salePrice;
    @ColumnInfo(name = "total_stock")
    private int totalStock;
    @ColumnInfo(name = "exp_date")
    private Date expDate;

    public Batch(long stockId, double originalPrice, double salePrice, int totalStock, Date expDate) {
        this.stockId = stockId;
        this.originalPrice = originalPrice;
        this.salePrice = salePrice;
        this.totalStock = totalStock;
        this.expDate = expDate;
    }

    public long getBatchId() {
        return batchId;
    }

    public void setBatchId(long batchId) {
        this.batchId = batchId;
    }

    public long getStockId() {
        return stockId;
    }

    public void setStockId(long stockId) {
        this.stockId = stockId;
    }

    public double getOriginalPrice() {
        return originalPrice;
    }

    public void setOriginalPrice(double originalPrice) {
        this.originalPrice = originalPrice;
    }

    public double getSalePrice() {
        return salePrice;
    }

    public void setSalePrice(double salePrice) {
        this.salePrice = salePrice;
    }

    public int getTotalStock() {
        return totalStock;
    }

    public void setTotalStock(int totalStock) {
        this.totalStock = totalStock;
    }

    public Date getExpDate() {
        return expDate;
    }

    public void setExpDate(Date expDate) {
        this.expDate = expDate;
    }

    @Override
    public String toString() {
        return "Batch{" +
                "batchId=" + batchId +
                ", productId=" + stockId +
                ", originalPrice=" + originalPrice +
                ", salePrice=" + salePrice +
                ", totalStock=" + totalStock +
                ", expDate=" + expDate +
                '}';
    }
}

File: app/src/main/java/com/example/myittaroostockinventorymanger/data/entities/Batch.kt
package com.example.myittaroostockinventorymanger.data.entities

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
@Entity
data class Batch(
    @ColumnInfo(name = "item_id")
    var itemId: Long,

    @ColumnInfo(name = "base_price")
    var basePrice: Double,

    @ColumnInfo(name = "selling_price")
    var sellingPrice: Double,

    @ColumnInfo(name = "quantity")
    var quantity: Int,

    @ColumnInfo(name = "exp_date")
    var expDate: Date
) : Parcelable {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "batch_id")
    var batchId: Long = 0

}
